api/plugins/bingo-elastic/python/README.md

```diff
@@ -0,0 +1,183 @@
+- [Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/install-elasticsearch.html)
+- many many more available on premise and as cloud products & services
+
+Something simple could be done as following:
+
+```
+docker run -p 9200:9200 --env "discovery.type=single-node" --env "opendistro_security.disabled=true" amazon/opendistro-for-elasticsearch:latest
+```
+
+### Usage 
+
+#### Create ElasticRepository
+
+```
+repository = ElasticRepository(host="127.0.0.1", port=9200)
+```
+
+Other customisations like SSL, custom number of shards/replicas, refresh interval, and many more are supported
+
+#### Read Indigo records from file
+
+IndigoRecord can be created from IndigoObject.
+ 
+Full usage example: 
+```
+from bingo_elastic.model.record import IndigoRecord
+from indigo import Indigo
+
+indigo = Indigo()
+compound = indigo.loadMoleculeFromFile("composition.mol")
+indigo_record = IndigoRecord(indigo_object=compound)
+```
+
+`bingo_elastic` provides helpers to load sdf, cml, smiles and smi files
+
+```
+from bingo_elastic.model import helpers
+
+sdf = helpers.iterate_sdf("compounds.sdf")
+cml = helpers.iterate_cml("compounds.cml")
+smi = helpers.iterate_smiles("compounds.smi")
+```
+
+Also function `helpers.iterate_file(file: Path)` is available. This function 
+selects correct iterate function by file extension. The `file` argument must 
+be `pathlib.Path` instance
+
+```
+from bingo_elastic.model import helpers
+from pathlib import Path
+
+sdf = helpers.iterate_file(Path("compounds.sdf"))
+```
+
+
+#### Index records into Elasticsearch
+
+Full usage example: 
+
+```
+from bingo_elastic.model import helpers
+from pathlib import Path
+
+repository = ElasticRepository(host="127.0.0.1", port=9200)
+sdf = helpers.iterate_file(Path("compounds.sdf"))
+repository.index_records(sdf);
+```
+
+*CAVEAT*: Elasticsearch doesn't have strict notion of commit, so records might appear in the index later on
+Read more about it here -  https://www.elastic.co/guide/en/elasticsearch/reference/master/index-modules.html#index-refresh-interval-setting
+
+For indexing one record the the method `ElasticRepository.index_record` can be used 
+
+#### Retrieve similar records from Elasticsearch
+
+```
+from bingo_elastic.predicates import TanimotoSimilarityMatch
+alg = TanimotoSimilarityMatch(target, 0.9)
+similar_records = repository.filter(similarity=alg, limit=20)
+```
+
+In this case we requested top-20 most similar molecules compared to `target` based on Tanimoto similarity metric
+
+Supported similarity algorithms:
+- `TanimotoSimilarityMatch`
+- `EuclidSimilarityMatch`
+- `TverskySimilarityMatch`
+- `ExactMatch`
+
+#### Find exact records from Elasticsearch
+
+```
+alg = TanimotoSimilarityMatch(target)
+exact_records = repository.filter(similarity=alg, limit=20)
+```
+
+In this case we requested top-20 candidate molecules with exact same fingerprint to `target`.
```

Comment:
default still 20, not 10 like in Java API?

Reply:
This limit explicitly set as argument `limit=20` 

I changed default limit to 10 in `ElasticRepository.filter` method